src/main/java/uk/gov/pay/api/exception/mapper/ViolationExceptionMapper.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.pay.api.exception.mapper;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.jersey.validation.JerseyViolationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.api.model.PaymentError;
+
+import javax.annotation.Priority;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static uk.gov.pay.api.model.PaymentError.Code.CREATE_PAYMENT_VALIDATION_ERROR;
+
+@Priority(1)
+public class ViolationExceptionMapper implements ExceptionMapper<JerseyViolationException> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViolationExceptionMapper.class);
+
+    @Override
+    public Response toResponse(JerseyViolationException exception) {
+        LOGGER.error(exception.getMessage());
+        ConstraintViolation<?> firstException = getFirstException(exception);
+        String message = firstException.getMessage();
+        String fieldName = getApiFieldName(firstException);
+        PaymentError paymentError = PaymentError.aPaymentError(fieldName, CREATE_PAYMENT_VALIDATION_ERROR, message);
+
+        return Response.status(422)
+                .entity(paymentError)
+                .build();
+    }
+
+    private String getApiFieldName(ConstraintViolation<?> firstException) {
+        Field field = getField(firstException);
+        return getJacksonPropertyName(field).orElse(field.getName());
```

Comment:
Why can't we simply return the `field.getName`; why do we have to get the jackson property name?

Reply:
That's because in the error message we want to use the name used in the api/json request and not just the name of the field on the class, e.g. returning `return_url` and not `returnUrl`. This will look for `@JsonProperty` e.g.  `@JsonProperty("return_url")` on the field which has failed validation and use that if it exists.

Reply:
👍